src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -131,6 +132,70 @@ export function fetchAll(
   }
 }
 
+export function fetchMoreDocuments(
+  dataSubjects: SavedSearchData,
+  fetchDeps: FetchDeps
+): Promise<void> {
+  const { getAppState, services, savedSearch } = fetchDeps;
+  const searchSource = savedSearch.searchSource.createChild();
+
+  try {
+    const lastDocuments = dataSubjects.documents$.getValue().result || [];
+    const lastDocumentSort = lastDocuments[lastDocuments.length - 1]?.raw?.sort;
+
+    if (!lastDocumentSort) {
+      return Promise.resolve();
+    }
+
+    searchSource.setField('searchAfter', lastDocumentSort);
+
+    const dataView = searchSource.getField('index')!;
+    const query = getAppState().query;
+    const recordRawType = getRawRecordType(query);
+
+    if (recordRawType === RecordRawType.PLAIN) {
+      // not supported yet
+      // TODO?
+      return Promise.resolve();
+    }
```

Comment:
I don't think for the initial version this is required, also I assume the logic would be very different, right?